### ISSUE TITLE
[Awaiting Checklist Completion] [$1000] Web - The  reveal /hide button text is cut off in offline mode and doesn't disappears dynamically

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#20946]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to staging dot on web chrome and login with User A (On another window login with User B)
2. From User A, send a message to User B and from User B mark the same message as assault
3. On User A, go offline and delete the message you sent earlier and notice that the button text is cut from the between. (Cutting the button text makes it look weird since we don't normally see any button texts that are cut from between)
4. You can see still the button reveal/hide is present. But now click on any other chat and return back to the previous chat and notice that the button disappears. Thus, the button should have dynamically disappeared when the user A deleted the message earlier.

## Expected Result:
The reveal /hide button text should not be cut off in the first place. Also, the button should have disappeared dynamically when the user has deleted the message in offline mode.

## Actual Result:
The  reveal /hide button text is cut off and doesn't disappear dynamically

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.3.27.3

**Reproducible in staging?:** yes

**Reproducible in production?:** yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/eed71b0a-2683-41b0-90f9-99c51bca44de

https://github.com/Expensify/App/assets/93399543/9062bcec-790c-48b1-8bf5-899b04722f82

**Expensify/Expensify Issue URL:**

**Issue reported by:** @priya-zha

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1686412725163069

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~014c79e15242357a09</li>
        <li>Upwork Job ID: 1670827258725896192</li>
        <li>Last Price Increase: 2023-07-03</li>
    </ul>
</details>

**Comment by twisterdotcom** [comment]
Okay, I agree. @dangrous - do you agree? This is pretty edge case but that's what open source bug reporting is for.

>The reveal /hide button text should not be cut off in the first place. Also, the button should have disappeared dynamically when the user has deleted the message in offline mode.

**Comment by dangrous** [comment]
So I think this is sorta two bugs, if I'm understanding correctly.
1. The button text is also given a ~strike through~ format when the comment is deleted. (Is that what you meant by "cut off" @kbecciv?). This is a bug and should be fixed, but I don't think is addressed in the existing proposal.
2. The button disappears when navigating away from a deleted message, while often. This is ALSO a bug, and @dukenv0307's proposal seems likely to fix it.

**Comment by dangrous** [comment]
yep I think that's it!

**Comment by ShogunFire** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

There are two problems when we delete a flagged comment offline:
1) The hide/reveal button disapears when we reopen the page and it should always show
2) The text of the button is striked through and it shouldn't
 
### What is the root cause of that problem?

1) As @dukenv0307 explained [here](https://github.com/Expensify/App/issues/20946#issuecomment-1595332987) it happens because we delete the moderationDecisions in Onyx in deleteReportComment function


2) The strike through is applied to every child of the comment in applyStrikeThrough function [here](https://github.com/Expensify/App/blob/d39643006874bf5ffec923291afa6b43884a4aec/src/components/OfflineWithFeedback.js#L77-L88) so even the text of the button is striked through

### What changes do you think we should make in order to solve the problem?


1) As @dukenv0307 proposed we can preserve the moderationdecision by adding 
`moderationDecisions: reportAction.message[0].moderationDecisions`
here:
https://github.com/Expensify/App/blob/d39643006874bf5ffec923291afa6b43884a4aec/src/libs/actions/Report.js#L789-L796

2) We can add a condition to only apply the strike through to certain childs, for example child that have a props action


Result:

https://github.com/Expensify/App/assets/19537677/d5b2efa9-8cb1-4e1f-999a-243508647072

### What alternative solutions did you explore? (Optional)

**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by twisterdotcom** [comment]
Always goes overdue on the weekend. @rushatgabhane will review soon.

**Comment by ShogunFire** [comment]
@rushatgabhane I don't see how we can know if we should show the button if we don't know if the comment had a moderateDecision.
For my proposal it is the same as @dukenv0307 for the first bug but I added the proposal for the strike through.

**Comment by dangrous** [comment]
Okay sooooo given these proposals - and given what would be needed to fix them - I'm wondering if we might want to change the expected behavior.

Since only the author could delete the message while offline, I don't have a huge problem with them seeing the message after they delete it. So should we - on delete - remove the flagged formatting entirely? Basically so it immediately looks like just a regular deleted message in all cases? i don't know why the author of the comment would want to be able to hide it.

What do you think, @twisterdotcom?

**Comment by twisterdotcom** [comment]
I think that makes sense yes.

**Comment by ShogunFire** [comment]
## Proposal
[Updated](https://github.com/Expensify/App/issues/20946#issuecomment-1605692842)

**Comment by twisterdotcom** [comment]
I am OOO! Adding another BZ for the week.

**Comment by conorpendergrast** [comment]
@rushatgabhane Ah, we need you to post a fresh comment with the double-ribbons! 
`:ribbon: :eyes: :ribbon:  C+ reviewed`

**Comment by MariaHCD** [comment]
@rushatgabhane is reviewing the [PR](https://github.com/Expensify/App/pull/22276).

**Comment by MariaHCD** [comment]
PR was merged today.

**Comment by twisterdotcom** [comment]
Payment summary:
- @dukenv0307 has been paid $1000 via Upwork here: https://www.upwork.com/nx/wm/workroom/34118657/overview
- @rushatgabhane will be paid $1000 via newDot
- @priya-zha is waiting on an offer for $250 here: https://www.upwork.com/nx/wm/offer/new/1010783361

**Comment by rushatgabhane** [comment]
made a manual request - https://staging.new.expensify.com/r/4074322401487931

**Comment by rushatgabhane** [comment]
WIP ⚠️ 

1. The PR that introduced the bug has been identified. Link to the PR:

2. The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:

3. A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

4. Determine if we should create a regression test for this bug.

5. If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again

**Comment by JmillsExpensify** [comment]
Reviewed details for @rushatgabhane. These details are accurate based on summary from Business Reviewer and are now approved for payment in NewDot.
